nixos/modules/config/users-groups.nix

```diff
@@ -568,5 +579,17 @@ in {
   imports =
     [ (mkAliasOptionModule [ "users" "extraUsers" ] [ "users" "users" ])
       (mkAliasOptionModule [ "users" "extraGroups" ] [ "users" "groups" ])
+      {
+        environment = {
+          etc = mapAttrs' (name: { packages, ... }: {
+            name = "per-user-pkgs/${name}";
+            value.source = pkgs.symlinkJoin {
+              name = "per-user-pkgs.${name}";
+              paths = packages;
+            };
+          }) cfg.users;
```

Comment:
Right. I've added a filter to only create symlinks where packages are defined. Though, I'm not sure about (silently) filtering out system users because it would be counterintuitive in case someone actually tries to define packages for them for some reason. We could add an assertion, but then I wouldn't want to patronize users.